### PR TITLE
Codespaces

### PR DESCRIPTION
**Pull request by CloCkWeRX**
Was https://github.com/Growstuff/growstuff/pull/3216

*Squash merge!*

### CODE REVIEW

File: .devcontainer/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      - xpack.security.http.ssl.certificate=certs/es01/es01.crt
+      - xpack.security.http.ssl.certificate_authorities=certs/ca/ca.crt
+      - xpack.security.http.ssl.verification_mode=certificate
+      - xpack.security.transport.ssl.enabled=true
+      - xpack.security.transport.ssl.key=certs/es01/es01.key
+      - xpack.security.transport.ssl.certificate=certs/es01/es01.crt
+      - xpack.security.transport.ssl.certificate_authorities=certs/ca/ca.crt
+      - xpack.security.transport.ssl.verification_mode=certificate
+      - xpack.license.self_generated.type=${LICENSE}
+    # mem_limit: ${MEM_LIMIT}
```
Comment (CloCkWeRX) [inline .devcontainer/docker-compose.yml]:
```suggestion
    mem_limit: ${MEM_LIMIT}
```

**Comment by CloCkWeRX** [comment]
Frustrating:
Container starts, you can curl elasticsearch.

Bundles.

You try to curl elasticsearch.

Name no longer resolves!

**Comment by CloCkWeRX** [comment]
Okay, restarting everything made it much more stable!

**Comment by CloCkWeRX** [comment]
Its. Working!

Try it out:
- Codespaces, start new codespace off of this branch.
- After waiting for it to build for ages, start your rails server and login!

image

I've updated the new contributors guide, and mucking around with VS Code locally seems to work nicely.

File: .devcontainer/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        condition: service_healthy
+      elasticsearch:
+        condition: service_healthy
+    environment:
+      ELASTICSEARCH_URL: http://elasticsearch:9200/
+      APP_DOMAIN_NAME: localhost:3000
+      APP_PROTOCOL: http
+      DEVISE_SECRET_KEY: secret
+      GROWSTUFF_EMAIL: "noreply@test.growstuff.org"
+      GROWSTUFF_FLICKR_KEY: secretkey"
```
Comment (cesy) [inline .devcontainer/docker-compose.yml:19]: Is there a missing or extra quote mark here?

**Review by cesy — APPROVED** [review]
Happy to merge, thank you so much for finishing this off.

File: .devcontainer/docker-compose.yml (hunk 2 above)
Comment (maco) [reply .devcontainer/docker-compose.yml:19]: wait, shouldn't merging have waited for an answer to this?

Comment (cesy) [reply .devcontainer/docker-compose.yml:19]: I figured it was minor enough that it doesn't matter so long as the overall solution works, or indeed if everything in a codespace works except Flickr, that's not worth stopping merging.

**Pull request merged**